features/facebookImport/src/facebookImporter.jsx

```diff
@@ -149,13 +150,17 @@ const FacebookImporterApp = () => {
             <ImporterProvider>
                 <PolyImportProvider
                     parentContext={ImporterContext}
-                    analyzeFile={analyzeFile}
                     dataImporters={dataImporters}
-                    subAnalyses={subAnalyses}
                     DataAccount={FacebookAccount}
                 >
-                    <div className="poly-nav-bar-separator-overlay" />
-                    <FacebookImporter />
+                    <PolyAnalysisProvider
+                        parentContext={PolyImportContext}
+                        subAnalyses={subAnalyses}
+                        analyzeFile={analyzeFile}
+                    >
+                        <div className="poly-nav-bar-separator-overlay" />
+                        <FacebookImporter />
```

Comment:
`<rant>`
The problem I see here is that you're forcing the clients of the providers to provide boilerplate for what is, essentially, the same code. Namely, you'll have to repeat this same code for GoogleImporter except this last line will be `<GoogleImporter />`. So instead of injecting the component at the top level, you're including it at the bottom level.
Main context for this rant is this documentation about [composition vs. inheritance](https://reactjs.org/docs/composition-vs-inheritance.html#containment). My take is that composition is preferred over inheritance. And the implication for this specific code is that, essentially, you would need to have `ImporterProvider` create the `importer` component that would be injected at the lowest level. No contexts would really be necessary, just passing props down in a (relatively) shallow hierarchy.
Please also note that you're seemingly using context as a prop, since you're passing it down as component props instead of defining those components as context consumers, maybe because it's complicated to define them as consumers of several different contexts. That way, you're kinda losing the advantages of having dynamic contexts, since contexts are being consumed only when components are invoked, and not when the context changes.
I don't know if all this makes sense. Baseline is, we're probably good with this as a first approximation. But all components are very tightly coupled, embedded over each other, and any refactoring is going to imply a good amount of changes up and down the nested context hierarchy.
`</rant>`

Reply:
I don't really understand what you mean. Contexts are basically components that live the composition dream. What do you mean? You wanna make a call?

Reply:
They are just regular consumers, but as the context is in a different package it needs to know what the parent is. It's not regular probs!

Reply:
> I don't really understand what you mean. Contexts are basically components that live the composition dream. What do you mean? You wanna make a call?

Maybe I'm not getting the whole context concept, so if you have time...

Reply:
I understand we wouldn't want to have the contexts too convoluted in the code but the only reason why the context is being passed as a prop is to avoid the need to import it in the package as they are separate packages. In regards to passing the variables of interest only rather than the full context, I think it could work but that would require more refactoring if we were to expand the poly-analysis context rather than just adding one more imported variable from the context.